src/ast.c

```diff
@@ -460,7 +462,10 @@ static value_t julia_to_scm_(jl_value_t *v)
         return scmv;
     }
     if (jl_typeis(v, jl_linenumbernode_type))
-        return julia_to_list2((jl_value_t*)line_sym, jl_fieldref(v,0));
+        return fl_cons(julia_to_scm_((jl_value_t*)line_sym),
+                       fl_cons(julia_to_scm_(jl_fieldref(v,1)),
+                       fl_cons(julia_to_scm_(jl_fieldref(v,0)), FL_NIL))
```

Comment:
i think you are missing flisp gc-roots for the intermediate `julia_to_scm_` and `fl_cons` calls (remember, C does not guarantee right-to-left execution within a statement so you need https://github.com/JuliaLang/julia/pull/12544/files#diff-ee48e36e4d097c33660f778b57339696R416)